From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: sebseb7 <sebgreenbus@gmail.com>
Date: Wed, 24 Jan 2024 04:14:28 +0100
Subject: [PATCH] WIP-mobSwitch-fixes


diff --git a/src/main/java/net/minecraft/world/entity/monster/Shulker.java b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
index ab21a44905a4154013cd65acdecbf55b741da086..c8bb2b4468d79e7cc5ee6ed354f9fd7c0f07c4e6 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
@@ -549,7 +549,7 @@ public class Shulker extends AbstractGolem implements VariantHolder<Optional<Dye
                         return;
                     }
                     // Paper end - Shulker duplicate event
-                    this.level().addFreshEntity(entityshulker, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.BREEDING); // CraftBukkit - the mysteries of life
+                    this.level().addFreshEntity(entityshulker, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.NATURAL); // CraftBukkit - the mysteries of life
                 }
 
             }
diff --git a/src/main/java/net/minecraft/world/level/NaturalSpawner.java b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
index 4d9e831f08cd3ed3765560964e88a412962a760b..140ca139a64d53cc24b6d64f996f39c482cce4d3 100644
--- a/src/main/java/net/minecraft/world/level/NaturalSpawner.java
+++ b/src/main/java/net/minecraft/world/level/NaturalSpawner.java
@@ -97,6 +97,7 @@ public final class NaturalSpawner {
                 // Paper start - Only count natural spawns
                 if (!entity.level().paperConfig().entities.spawning.countAllMobsForSpawning &&
                     !(entity.spawnReason == org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.NATURAL ||
+                        entity.spawnReason == org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.INFECTION ||
                         entity.spawnReason == org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.CHUNK_GEN)) {
                     continue;
                 }
